crates/red_knot/src/module.rs

```diff
@@ -611,6 +670,10 @@ where
 {
     let mut package_path = module_search_path.path().to_path_buf();
 
+    if module_search_path.kind().is_standard_library() {
+        package_path = package_path.join(TYPESHED_STDLIB_DIRECTORY);
+    }
```

Comment:
I guess this is so when people point to a custom typeshed, they can just point to the root of the typeshed repo?

What about instead resolving this once upfront in the way we set our stdlib and typeshed-third-party search paths, so we don't have to do it on every resolution?

Reply:
> I guess this is so when people point to a custom typeshed, they can just point to the root of the typeshed repo?

Yup. That's what mypy and pyright do currently, and it's what I would expect a `--custom-typeshed-dir` argument to do.

> What about instead resolving this once upfront in the way we set our stdlib and typeshed-third-party search paths, so we don't have to do it on every resolution?

Could do! I decided to go this way so that calling `.path()` on the `ModuleSearchPath` instance would give you the path for `typeshed` rather than the path for `typeshed/stdlib`. That felt like it made the most sense here. As discussed elsewhere, I think we're going to have to do different things for each different kind of module search path anyway, so we're not going to be able to treat all paths in the resolved vector the same way even if this module search path pointed to `typeshed/stdlib` rather than just `typeshed`.

Reply:
To me it seems more useful if `.path()` on the `ModuleSearchPath` instance points to the actual path imports will be resolved from.

And agreed that `foo-stubs` packages mean special handling for resolutions in site-packages specifically, but in general the more work we can do upfront and the less work we can do on resolution of each package, the better.

Reply:
Okay, no strong opinion from me on this; I can change this!